### PR TITLE
Edit profile  and sign up functionality completed

### LINKED ISSUE #9: Sign Up
# REQUIREMENTS
- Sign Up with: Name, email, password,  Birthday,  gender, 
- Add user to database

# Acceptance Criteria
List all required acceptance criteria
- [ ] Database get a new record with the sign up form

### PR DESCRIPTION
**Pull request by IrenehMDev**
Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>

# Description
- Added last_name field to the user table from migration.
- Added required fields to Sign Up's view (Devise).
- Added required fields to Edit Profile's view (Devise) like to Sign Up's view.
- Implemented just sign up and edit profile functionality.
- Generated Devise Controllers.
 
  ## Screenshots 
image

image

image

image

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+    before_action :configure_permitted_parameters, if: :devise_controller?
+
+
+    protected
+
+        def configure_permitted_parameters
+            devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :last_name, :email, :password, :birthday, :gender) }
+            devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:name, :last_name, :email, :password, :current_password, :birthday, :gender) }
+        
+        end
```
Comment (JuanequeX) [inline app/controllers/application_controller.rb]: Please review this indentation, two spaces are the ideal

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  # def cancel
+  #   super
+  # end
+
+  # protected
+
+  # If you have extra params to permit, append them to the sanitizer.
+   def configure_sign_up_params
+     devise_parameter_sanitizer.permit(:sign_up, keys: [:name, :last_name, :birthday, :gender])
+  # end
```
Comment (JuanequeX) [inline app/controllers/users/registrations_controller.rb]: This method cannot be useful if you also comment the `end` xd

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/controllers/users/confirmations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::ConfirmationsController < Devise::ConfirmationsController
+  # GET /resource/confirmation/new
```
Comment (kevinnio) [inline app/controllers/users/confirmations_controller.rb]: Is there a reason to have all these comments here? What do you think about removing them?

File: app/views/devise/registrations/new.html.erb [hunk 4]
```diff
@@ -21,6 +31,18 @@
...
 
+  <div class="field">
+    <%= f.label :birthday %><br />
+    <%= f.date_field :birthday, autofocus: true %>
+  </div>
+
+  <div class="control-group">
+    <%= f.label "Gender", class: 'control-label' %><br />
+    <div class="controls">
+    <%= f.select(:gender, options_for_select([['Female','Female'],['Male','Male']])) %>
```
Comment (kevinnio) [inline app/views/devise/registrations/new.html.erb]:
```suggestion
      <%= f.select(:gender, options_for_select([['Female','Female'],['Male','Male']])) %>
```

File: app/views/devise/registrations/edit.html.erb [hunk 5]
```diff
@@ -31,6 +41,18 @@
...
 
+  <div class="field">
+    <%= f.label :birthday %><br />
+    <%= f.date_field :birthday, autofocus: true %>
+  </div>
+
+  <div class="control-group">
+    <%= f.label "Gender", class: 'control-label' %><br />
+    <div class="controls">
+    <%= f.select(:gender, options_for_select([['Female','Female'],['Male','Male']])) %>
```
Comment (kevinnio) [inline app/views/devise/registrations/edit.html.erb]:
```suggestion
      <%= f.select(:gender, options_for_select([['Female','Female'],['Male','Male']])) %>
```

File: app/controllers/users/registrations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class Users::RegistrationsController < Devise::RegistrationsController
+   before_action :configure_sign_up_params, only: [:create]
+  # before_action :configure_account_update_params, only: [:update]
+
+  # GET /resource/sign_up
+   def new
+     super
```
Comment (kevinnio) [inline app/controllers/users/registrations_controller.rb]: Is there any reason to define `#new` here? If this methods just calls `super`, then it's the same as just using the inherited method from `Devise::RegistrationsController`. 🤔

**Review by kevinnio — CHANGES_REQUESTED** [review]

**Review by kevinnio — APPROVED** [review]

**Review by YilKanda — APPROVED** [review]

**Review by DerekAyala — APPROVED** [review]

**Review by JuanequeX — APPROVED** [review]
LGTM

**Pull request merged**